src/node/internal/internal_http_outgoing.ts

```diff
@@ -474,6 +475,12 @@ export class OutgoingMessage extends Writable implements _OutgoingMessage {
 
   // @ts-expect-error TS2611 Required for accessor
   get writableLength(): number {
+    // If using buffer with headers, include buffer's writable length
+    if (this.#buffer != null && (this._header !== null || this._headerSent)) {
+      return (
+        this.outputSize + this.#buffer.writableLength + this[kChunkedLength]
+      );
+    }
```

Comment:
Is this the same behaviour as node? (Usually I can find the corresponding bit of the code but my search is failing.)

Reply:
Yes, tests are validating it.